src/Persistence/SitelinkPageItemLookup.php

```diff
@@ -18,7 +18,7 @@ public function __construct(
 	}
 
 	public function getItemId( Title $title ): ?ItemId {
-		return $this->sitelinkLookup->getItemIdForLink( $this->sitelinkSiteId, $title->getText() );
+		return $this->sitelinkLookup->getItemIdForLink( $this->sitelinkSiteId, $title->getPrefixedText() );
```

Comment:
Not related to this PR, but it seems like we overcomplicated things with the `PageItemLookup` implementations. We have NullPageItemLookup and SitelinkPageItemLookup constructed via PageItemLookupFactory. These 3 can be combined into one implementation of PageItemLookup without any design downsides and some upsides.

Reply:
Tracked by https://github.com/ProfessionalWiki/WikibaseFacetedSearch/issues/246